src/test/java/de/tum/cit/aet/artemis/programming/util/ProgrammingExerciseImportTestService.java

```diff
@@ -0,0 +1,167 @@
+        }
+
+        Object additionalData = modifier.modify(parsedExercise);
+
+        parsedExercise.setCourse(course);
+        parsedExercise.setId(null);
+        parsedExercise.setChannelName("testchannel-pe-imported");
+        parsedExercise.forceNewProjectKey();
+
+        MockMultipartFile file = new MockMultipartFile("file", "test.zip", "application/zip", resource.getInputStream());
+
+        ProgrammingExercise importedExercise = request.postWithMultipartFile("/api/programming/courses/" + course.getId() + "/programming-exercises/import-from-file",
+                parsedExercise, "programmingExercise", file, ProgrammingExercise.class, HttpStatus.OK);
+
+        return new ImportFileResult(resource, parsedExercise, importedExercise, additionalData);
+    }
+
+    /**
+     * Counts occurrences of a string in a zip file
+     *
+     * @param resource     Resource containing the zip file
+     * @param searchString String to search for
+     * @return Count of occurrences
+     * @throws Exception if the zip file cannot be read
+     */
+    public int countOccurrencesInZip(ClassPathResource resource, String searchString) throws Exception {
+        int count = 0;
+        try (ZipInputStream zipIn = new ZipInputStream(new FileInputStream(resource.getFile()))) {
+            ZipEntry zipEntry;
+            while ((zipEntry = zipIn.getNextEntry()) != null) {
+                if (!zipEntry.isDirectory() && zipEntry.getName().endsWith(".zip")) {
+                    ByteArrayOutputStream baos = new ByteArrayOutputStream();
+                    byte[] buf = new byte[1024];
+                    int n;
+                    while ((n = zipIn.read(buf)) > 0) {
+                        baos.write(buf, 0, n);
+                    }
+                    String content = baos.toString(StandardCharsets.UTF_8);
+                    int idx = 0;
+                    while ((idx = content.indexOf(searchString, idx)) != -1) {
+                        count++;
+                        idx += searchString.length();
+                    }
+                }
+            }
+        }
+        return count;
+    }
+
+    /**
+     * Counts occurrences of a string in files within a directory
+     *
+     * @param path         Directory path to search in
+     * @param searchString String to search for
+     * @return Count of occurrences
+     * @throws IOException if the directory cannot be read
+     */
+    public int countOccurrencesInDirectory(Path path, String searchString) throws IOException {
+        int count = 0;
+        if (!Files.exists(path)) {
+            throw new IOException("Directory does not exist");
+        }
+        List<Path> files = new ArrayList<>();
+        Files.walk(path).filter(Files::isRegularFile).forEach(files::add);
+        for (Path filePath : files) {
+            String content = new String(Files.readAllBytes(filePath), StandardCharsets.UTF_8);
```

Comment:
Can't we achieve this with Files.readString? 
```suggestion
            String content = Files.readString(filePath);
```

Reply:
Current version of the code fails with Files.readString. Next time I will try to start with Files.readString from scratch 👍